### PR TITLE
Linked list insertions

### PR DESCRIPTION
**Pull request by eden-brekke**
- [x] Top-level README “Table of Contents” is updated
 - [x] README for this challenge is complete
 - [x] Summary, Description, Approach & Efficiency, Solution
 - [x] Picture of whiteboard
 - [x] Link to code
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
 - [x] “Happy Path” - Expected outcome
 - [x] Expected failure
 - [x] Edge Case (if applicable/obvious)

### CODE REVIEW

File: python/data_structures/linked_list.py [hunk 1]
```diff
@@ -34,15 +34,64 @@ def includes(self, value):
...
+        else:
+            self.head = new_node
+        return new_node
+
+    def insert_before(self, value, new_value):
+        previous = current = self.head
+        if self.head is None:
+            raise TargetError
+        if self.includes(value) is False:
+            raise TargetError
```
Comment (jstreifel-33) [inline python/data_structures/linked_list.py:53]: There's a case to be made that lines 50 and 51 here are unnecessary. Looking back at your .includes method, the condition on 52 should adequately cover a case where self.head is None.
